explorer/deploy/webAppCreator.ts

```diff
@@ -408,13 +420,22 @@ class WebsiteStep extends WebAppCreatorStepBase {
     private _serverUserName: string;
     private _serverPassword: string;
     private _imageName: string;
+    private _imageSubscription: Subscription;
+    private _registry: Registry;
 
     constructor(wizard: WizardBase, azureAccount: AzureAccountWrapper, context: AzureImageNode | DockerHubImageNode) {
         super(wizard, 'Create Web App', azureAccount);
 
         this._serverUrl = context.serverUrl;
-        this._serverPassword = context.password;
-        this._serverUserName = context.userName;
+        if (context instanceof DockerHubImageNode) {
+            this._serverPassword = context.password;
+            this._serverUserName = context.userName;
+        }
+        if (context instanceof AzureImageNode) {
```

Comment:
>if [](start = 8, length = 2)

`else if` to avoid additional check

Reply:
you also might to log if context is neither DockerHubImageNode or AzureImageNode

---
In reply to: [206671989](https://github.com/Microsoft/vscode-docker/pull/341#discussion_r206671989) [](ancestors = 206671989)